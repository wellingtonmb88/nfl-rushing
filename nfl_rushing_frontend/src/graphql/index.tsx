
import { gql } from 'apollo-boost';


export const FILTER_PLAYERS_BY_NAME = gql`
  query FilterPalyersByName($name: String, $skip: Int, $take: Int) {
    filterPalyersByName(name: $name, skip: $skip, take: $take) {
      total
      items {
      name
      team
      position
      longestRush
      rushingAttempts
      rushingAttemptsPerGame
      rushingAvgYardsPerAttempt
      rushingFirstDowns
      rushingFirstDownsPercentage
      rushingFortyPlusYeards
      rushingFumbles
      rushingTwentyPlusYeards
      rushingYardsPerGame
      totalRushingTouchdowns
      totalRushingYards
      }
   }
 }
`;

export const ALL_PLAYERS = gql`
  query Players {
    players {
      total
      items {
      name
      team
      position
      longestRush
      rushingAttempts
      rushingAttemptsPerGame
      rushingAvgYardsPerAttempt
      rushingFirstDowns
      rushingFirstDownsPercentage
      rushingFortyPlusYeards
      rushingFumbles
      rushingTwentyPlusYeards
      rushingYardsPerGame
      totalRushingTouchdowns
      totalRushingYards
      }
   }
 }
`;