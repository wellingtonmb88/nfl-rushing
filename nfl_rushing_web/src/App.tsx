import React from 'react';
import _ from 'lodash';
import Table from './components/Table'
import TextField from '@material-ui/core/TextField';
import AppContext from './context';
import CSVButtons from './components/CSVButtons'
import './styles/App.css';

function App() {
  const search = _.debounce((value: string, func: (text: string) => void) => {
    func(value);
  }, 200);

  return (
    <AppContext.Consumer>
      {({ searchPlayersByName, currentSearch }) => (
        <div className="App">
          <TextField id="standard-basic" label="Player Name"
            onChange={(props) => {
              const { value } = props.target;
              search(value, (text) => {
                searchPlayersByName(text, 0, 0);
              });
            }
            } />
          <CSVButtons />
          <Table currentSearchName={currentSearch} />
        </div>)
      }
    </AppContext.Consumer>
  );
}


export default App;
