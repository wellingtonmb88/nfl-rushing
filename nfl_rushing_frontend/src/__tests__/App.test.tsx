import React from 'react';
import { create } from 'react-test-renderer';
import App from '../App';
import { MockedProvider } from "@apollo/react-testing";
import { FILTER_PLAYERS_BY_NAME } from '../graphql';


const mocks = [
  {
    request: {
      query: FILTER_PLAYERS_BY_NAME,
      variables: { name: "Adam", skip: 0, take: 5}
    },
    result: {
      "data": {
        "filterPalyersByName": {
          "items": [
            {
              "longestRush": "11",
              "name": "Adam Thielen",
              "position": "WR",
              "rushingAttempts": 2,
              "rushingAttemptsPerGame": 0.1,
              "rushingAvgYardsPerAttempt": 7.5,
              "rushingFirstDowns": 1,
              "rushingFirstDownsPercentage": 50,
              "rushingFortyPlusYeards": 0,
              "rushingFumbles": 0,
              "rushingTwentyPlusYeards": 0,
              "rushingYardsPerGame": 0.9,
              "team": "MIN",
              "totalRushingTouchdowns": 0,
              "totalRushingYards": 15
            },
            {
              "longestRush": "8",
              "name": "Adam Humphries",
              "position": "WR",
              "rushingAttempts": 5,
              "rushingAttemptsPerGame": 0.3,
              "rushingAvgYardsPerAttempt": 3.6,
              "rushingFirstDowns": 0,
              "rushingFirstDownsPercentage": 0,
              "rushingFortyPlusYeards": 0,
              "rushingFumbles": 0,
              "rushingTwentyPlusYeards": 0,
              "rushingYardsPerGame": 1.2,
              "team": "TB",
              "totalRushingTouchdowns": 0,
              "totalRushingYards": 18
            }
          ],
          "total": 326
        }
      }
    }
  }
]

test('renders App', () => {
  const component = create(<MockedProvider mocks={mocks}><App /></MockedProvider>);
  expect(component.toJSON()).toMatchSnapshot();
});
