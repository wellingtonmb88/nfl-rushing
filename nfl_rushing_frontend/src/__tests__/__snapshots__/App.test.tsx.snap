// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
<div
  className="App"
>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
      htmlFor="standard-basic"
      id="standard-basic-label"
    >
      Player Name
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="standard-basic"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
  </div>
  <div
    className="makeStyles-buttonGroup-62"
  >
    <div
      aria-label="text primary button group"
      className="MuiButtonGroup-root"
      role="group"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedText MuiButtonGroup-groupedTextHorizontal MuiButtonGroup-groupedTextPrimary MuiButton-textPrimary Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Export CSV (Selected Items)
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedText MuiButtonGroup-groupedTextHorizontal MuiButtonGroup-groupedTextPrimary MuiButton-textPrimary Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Export CSV (Current Page)
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedText MuiButtonGroup-groupedTextHorizontal MuiButtonGroup-groupedTextPrimary MuiButton-textPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Export CSV (All Players Data)
        </span>
      </button>
    </div>
  </div>
  <div
    className="makeStyles-root-58"
  >
    <div
      className="MuiPaper-root makeStyles-paper-60 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular makeStyles-root-147 MuiToolbar-gutters"
      >
        <h6
          className="MuiTypography-root makeStyles-title-149 MuiTypography-h6"
          id="tableTitle"
        >
          NFL Rushing Players
        </h6>
      </div>
      <div
        className="MuiTableContainer-root"
      >
        <table
          aria-label="enhanced table"
          aria-labelledby="tableTitle"
          className="MuiTable-root makeStyles-table-61"
        >
          <thead
            className="MuiTableHead-root"
          >
            <tr
              className="MuiTableRow-root MuiTableRow-head"
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-paddingCheckbox"
                scope="col"
              >
                <span
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-211 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={null}
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <input
                      aria-label="select all players"
                      checked={false}
                      className="PrivateSwitchBase-input-214"
                      data-indeterminate={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                </span>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-paddingNone"
                scope="col"
              >
                Player
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                scope="col"
              >
                Team
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                scope="col"
              >
                Pos
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                <span
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiTableSortLabel-root"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  Lng
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                Att
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                Att/G
              </th>
              <th
                aria-sort="ascending"
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                <span
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiTableSortLabel-root MuiTableSortLabel-active"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  Yds
                  <span
                    className="makeStyles-visuallyHidden-63"
                  >
                    sorted ascending
                  </span>
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                Avg
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                Yds/G
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                <span
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiTableSortLabel-root"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  TD
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                1st
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                1st%
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                20+
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                40+
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                FUM
              </th>
            </tr>
          </thead>
          <tbody
            className="MuiTableBody-root"
          />
        </table>
      </div>
      <div
        className="MuiTablePagination-root"
      >
        <div
          className="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
        >
          <div
            className="MuiTablePagination-spacer"
          />
          <p
            className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
          >
            Rows per page:
          </p>
          <div
            className="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
            onClick={[Function]}
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby=" "
              className="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              5
            </div>
            <input
              onAnimationStart={[Function]}
              type="hidden"
              value={5}
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
          >
            0-0 of 0
          </p>
          <div
            className="MuiTablePagination-actions"
          >
            <button
              aria-label="Previous page"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Previous page"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="Next page"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Next page"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
